command/action_test.go

```diff
@@ -1,58 +1,74 @@
 package command
 
 import (
-	"github.com/cyberark/summon/secretsyml"
-	. "github.com/smartystreets/goconvey/convey"
-	_ "golang.org/x/net/context"
+	"bytes"
+	"errors"
+	"io"
 	"io/ioutil"
 	"os"
 	"os/exec"
 	"path"
 	"strings"
 	"testing"
-	"errors"
+
+	"github.com/cyberark/summon/secretsyml"
+	. "github.com/smartystreets/goconvey/convey"
+	_ "golang.org/x/net/context"
 )
 
 func TestRunAction(t *testing.T) {
 	Convey("Using a dummy provider that returns 'mysecret'", t, func() {
 		providerPath := path.Join(os.Getenv("PWD"), "testprovider.sh")
 
 		Convey("Passing in secrets.yml via --yaml", func() {
-			out, err := runAction(&ActionConfig{
-				Args:       []string{"printenv", "MYVAR"},
-				Provider:   providerPath,
-				Filepath:   "",
-				YamlInline: "MYVAR: !var somesecret/on/a/path",
-				Subs:       map[string]string{},
-				Ignores:    []string{},
+			var err error
+
+			output := captureStdout(func() {
+				err = runAction(&ActionConfig{
+					Args:       []string{"printenv", "MYVAR"},
+					Provider:   providerPath,
+					Filepath:   "",
+					YamlInline: "MYVAR: !var somesecret/on/a/path",
+					Subs:       map[string]string{},
+					Ignores:    []string{},
+				})
+
 			})
-			So(out, ShouldEqual, "mysecret\n")
+
 			So(err, ShouldBeNil)
+			So(output, ShouldEqual, "mysecret\n")
 		})
 
 		Convey("Errors when fetching keys return error", func() {
-			_, err := runAction(&ActionConfig{
+			err := runAction(&ActionConfig{
 				Args:       []string{"printenv", "MYVAR"}, // args
 				Provider:   providerPath,                  // provider
 				Filepath:   "",                            // filepath
 				YamlInline: "MYVAR: !var error",           // yaml inline
 				Subs:       map[string]string{},           // subs
 				Ignores:    []string{},                    // ignore
 			})
+
 			So(err, ShouldNotBeNil)
```

Comment:
An assertion here on the error message probably would have caught the format bug above.

Reply:
Looks like you added this, thanks!

`So(err.Error(), ShouldEqual, "Error fetching variable MYVAR: exit status 1")`